File: board-room/src/main/java/net/stzups/board/server/websocket/protocol/client/ClientMessageHandshake.java
package net.stzups.board.server.websocket.protocol.client;

public class ClientMessageHandshake extends ClientMessage {
    private long token;

    public ClientMessageHandshake(long token) {
        super(ClientMessageType.HANDSHAKE);
        this.token = token;
    }

    public long getToken() {
        return token;
    }
}

File: board-frontend/html/scripts/protocol/client/messages/ClientMessageHandshake.js
import ClientMessageType from "../ClientMessageType.js";
import ClientMessage from "../ClientMessage.js";

export default class ClientMessageHandshake extends ClientMessage {
    constructor(id, token) {
        super(ClientMessageType.HANDSHAKE);
        this.id = id;
        this.token = token;
    }

    getBufferSize() {
        return super.getBufferSize() + 16;
    }

    serialize(writer) {
        super.serialize(writer);
        writer.writeBigInt64(this.id);
        writer.writeBigInt64(this.token);
    }
}